### ISSUE TITLE
[question] How do i change the shortcut keys `ctrl+space` to other keys?

### ISSUE THREAD

**Issue by R32** [fdorg/flashdevelop#1298]
Because `Ctrl + space` has been occupied by system IME。

**Comment by Neverbirth** [comment]
Wow, really? cannot you change it? In Tools -> Shortcuts, it's called ASCompletion.CtrlSpace.

**Comment by R32** [comment]
But I can't find any that are associated with ASCompletion.CtrlSpace or ctrl + space

**Comment by Neverbirth** [comment]
Yeah, just checked, the shortcut is not configurable in FD at the moment. When I replied I seemed to recall so, but I saw that ASCompletion.CtrlSpace reference and thought that maybe I was wrong.

Too bad. It can be fixed, but it's more than just setting it there, plus there is the Ctrl Space handling in other contexts.

**Comment by wise0704** [comment]
The `Ctrl+Space`, `Ctrl+Shift+Space`, `Ctrl+Alt+Space` and `Ctrl+Alt+Shift+Space` combinations are all hardcoded, and it will take some time to make it configurable.
I intended to include it in the advanced shortcuts, but I ended up not touching it because I kind of found it unrelated to the aim of the PR.

**Comment by wise0704** [comment]
Side note: `Ctrl+Space` never worked for me to change IME. `Win+Space` works for changing input language.

**Comment by elsassph** [comment]
Although the hardcoded one persists, there should already be a way to set an alternate shortcut to trigger `Ctrl+Space`. I think that's `ASCompletion.CtrlSpace`; isn't it working anymore?

**Comment by Neverbirth** [comment]
ASCompletion.CtrlSpace is not hooked at all, it's not even registered, so it only exists in comments of the source code. I don't remember seeing it at all any of the times I had to touch the related parts of code :S.

**Comment by wise0704** [comment]
Those hardcoded shortcuts are better if changed to modifiable, as they're the only ones registered in `IgnoredKeys`, and rather create more mess in the shortcut mechanism.
I could work on this while I'm working on contextual shortcuts, as that will have to shake up many shortcut-related things anyway. Don't expect that to be happening anytime soon though.
